subsys/bluetooth/audio/mcs.c

```diff
@@ -909,120 +1130,170 @@ static void mark_icon_url_changed_cb(struct bt_conn *conn, void *data)
 {
-	LOG_DBG_OBJ_ID("Notifying current track ID: ", id);
-	notify(BT_UUID_MCS_CURRENT_TRACK_OBJ_ID, &id, BT_OTS_OBJ_ID_SIZE);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_CURRENT_TRACK_OBJ_ID_CHANGED));
 }
 
 void media_proxy_sctrl_next_track_id_cb(uint64_t id)
 {
-	if (id == MPL_NO_TRACK_ID) {
-		/* "If the media player has no next track, the length of the */
-		/* characteristic shall be zero." */
-		LOG_DBG_OBJ_ID("Notifying EMPTY next track ID: ", id);
-		notify(BT_UUID_MCS_NEXT_TRACK_OBJ_ID, NULL, 0);
-	} else {
-		LOG_DBG_OBJ_ID("Notifying next track ID: ", id);
-		notify(BT_UUID_MCS_NEXT_TRACK_OBJ_ID, &id, BT_OTS_OBJ_ID_SIZE);
-	}
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_NEXT_TRACK_OBJ_ID_CHANGED));
 }
 
 void media_proxy_sctrl_parent_group_id_cb(uint64_t id)
 {
-	LOG_DBG_OBJ_ID("Notifying parent group ID: ", id);
-	notify(BT_UUID_MCS_PARENT_GROUP_OBJ_ID, &id, BT_OTS_OBJ_ID_SIZE);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_PARENT_GROUP_OBJ_ID_CHANGED));
 }
 
 void media_proxy_sctrl_current_group_id_cb(uint64_t id)
 {
-	LOG_DBG_OBJ_ID("Notifying current group ID: ", id);
-	notify(BT_UUID_MCS_CURRENT_GROUP_OBJ_ID, &id, BT_OTS_OBJ_ID_SIZE);
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_CURRENT_GROUP_OBJ_ID_CHANGED));
 }
+#endif /* CONFIG_BT_OTS */
 
 void media_proxy_sctrl_playing_order_cb(uint8_t order)
 {
-	LOG_DBG("Notifying playing order: %d", order);
-	notify(BT_UUID_MCS_PLAYING_ORDER, &order, sizeof(order));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_PLAYING_ORDER_CHANGED));
 }
 
 void media_proxy_sctrl_media_state_cb(uint8_t state)
 {
-	LOG_DBG("Notifying media state: %d", state);
-	notify(BT_UUID_MCS_MEDIA_STATE, &state, sizeof(state));
+	bt_conn_foreach(BT_CONN_TYPE_LE, defer_value_ntf,
+			UINT_TO_POINTER(FLAG_MEDIA_STATE_CHANGED));
+}
+
+static void defer_media_control_point_ntf(struct bt_conn *conn, void *data)
+{
+	struct client_state *client = &clients[bt_conn_index(conn)];
+	const struct mpl_cmd_ntf *cmd_ntf = data;
+	struct bt_conn_info info;
+	int err;
+
+	err = bt_conn_get_info(conn, &info);
+	if (err != 0) {
+		LOG_ERR("Failed to get conn info: %d", err);
+		return;
+	}
+
+	if (info.state != BT_CONN_STATE_CONNECTED) {
+		/* Not connected */
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(client->flags, FLAG_MEDIA_CONTROL_POINT_BUSY)) {
+		client->cmd_ntf = *cmd_ntf;
+		atomic_set_bit(client->flags, FLAG_MEDIA_CONTROL_POINT_RESULT);
+		k_work_submit(&deferred_nfy_work);
+	}
 }
 
 void media_proxy_sctrl_command_cb(const struct mpl_cmd_ntf *cmd_ntf)
 {
-	LOG_DBG("Notifying control point command - opcode: %d, result: %d",
-		cmd_ntf->requested_opcode, cmd_ntf->result_code);
-	notify(BT_UUID_MCS_MEDIA_CONTROL_POINT, cmd_ntf, sizeof(*cmd_ntf));
+	/* FIXME: Control Point notification shall be sent to operation initiator only */
```

Comment:
Does this also apply for the search control point? 

How much effort is there to fix this in this PR?